### PR TITLE
br: Create grpc.connection with block mode

### PR DESCRIPTION
**Pull request by joccau**
Signed-off-by: joccau <zak.zhao@pingcap.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/29562

Problem Summary:  
It get timeout error when  restore  after 1 seconds, but 1 second is not up to timeout (5s)

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by YuJuncen — APPROVED** [review]

**Comment by joccau** [comment]
@YuJuncen  So quickly ~  Good~

**Comment by YuJuncen** [comment]
@joccau BTW, I am still wondering why make it block fix the problem. I guess it would be better to link #29562 in the comment for tracing🤔.

File: br/pkg/restore/client.go [hunk 1]
```diff
@@ -754,6 +754,7 @@ func (rc *Client) switchTiKVMode(ctx context.Context, mode import_sstpb.SwitchMo
 			gctx,
 			store.GetAddress(),
 			opt,
+			grpc.WithBlock(),
```
Comment (kennytm) [inline br/pkg/restore/client.go:757]: should we also include [`grpc.FailOnNonTempDialError(true)`](https://pkg.go.dev/google.golang.org/grpc@v1.29.0#FailOnNonTempDialError)

**Review by kennytm — APPROVED** [review]

**Comment by joccau** [comment]
/needs-cherry-pick-5.3.1

**Comment by joccau** [comment]
needs-cherry-pick-5.3.1

**Pull request merged**